mmeval/metrics/bleu.py

```diff
@@ -0,0 +1,193 @@
+
+    Args:
+        sentence(str): A sentence.
+
+    Returns:
+        Sequence[str]:A Sequence of tokens after word segmentation.
+    """
+    return sentence.split()
+
+
+def _get_brevity_penalty(pred_len: np.array,
+                         references_len: np.array) -> np.array:
+    """This function is used to calculate penalty factor.
+
+    Args:
+        pred_len(np.array): number of grams in the predicted sentence.
+        references_len(np.array): number of grams in the references.
+
+    Returns:
+        np.array: penalty factor.
+    """
+    if pred_len > references_len:
+        return np.array(1.)
+    return np.array(np.exp(1 - references_len / pred_len))
+
+
+class Bleu(BaseMetric):
+    """Bilingual Evaluation Understudy metric.
+
+       This metric is a tool for evaluating the quality of machine translation.
+       The closer the translation is to human translation,
+       the higher the score will be.
+
+       Args:
+           n_gram (int):The maximum number of words contained in a phrase
+               when calculating word fragments. Defaults to 4.
+           smooth(bool): Whether or not to apply smoothing. Default to False.
+           ngram_weights(optional | Sequence[float]): Weights used
+               for unigrams, bigrams, etc. to calculate BLEU score.
+               If not provided, uniform weights are used.Default to None.
+           **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+       Examples:
+
+            >>> predictions = ['the cat is on the mat','There is a big tree near the park here'] # type: ignore  # noqa: E501
+            >>> references = [['a cat is on the mat'],['A big tree is growing near the park here']] # type: ignore # noqa: E501
+            >>> bleu = Bleu()
+            >>> bleu_results = bleu(predictions, references)
+            {'bleu': ...}
+
+        Calculate Bleu with smooth:
+
+            >>> predictions = ['the cat is on the mat','There is a big tree near the park here'] # type: ignore # noqa: E501
+            >>> references = [['a cat is on the mat'],['A big tree is growing near the park here']] # type: ignore # noqa: E501
+            >>> bleu = Bleu(smooth = True)
+            >>> bleu_results = bleu(predictions, references)
+            {'bleu': ...}
+    """
+
+    def __init__(self,
+                 n_gram: int = 4,
+                 smooth: bool = False,
+                 ngram_weights: Optional[Sequence[float]] = None,
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.n_gram = n_gram
+        self.smooth = smooth
+        if ngram_weights is not None and len(ngram_weights) != n_gram:
+            raise ValueError(
+                f'List of weights has different weights than `n_gram`: '
+                f'{len(ngram_weights)} != {n_gram}')
+        self.ngram_weights = ngram_weights if ngram_weights is not None else [
+            1.0 / n_gram
+        ] * n_gram
```

Comment:
Suggestion that improve code readability~
```suggestion
        if ngram_weights is None:
            ngram_weights = [ 1.0 / n_gram ] * n_gram
        self.ngram_weights = ngram_weights
```